lib/group_repository.py

```diff
@@ -25,6 +28,60 @@
 logger = get_logger(__name__)
 
 
+def add_hosts_to_group(group_id: str, host_id_list: List[str]):
+    current_org_id = get_current_identity().org_id
+
+    # Check if the hosts are already associated with another group
+    assoc_query = HostGroupAssoc.query.filter(HostGroupAssoc.host_id.in_(host_id_list)).all()
+    if assoc_query:
+        taken_hosts = [assoc.host_id for assoc in assoc_query]
+        log_host_group_add_failed(logger, taken_hosts, group_id)
+        raise InventoryException(
+            title="Invalid request",
+            detail=f"The following subset of hosts are already associated with another group: {taken_hosts}.",
+        )
+
+    # Check if any hosts in the list don't belong to the current org_id
+    host_query = Host.query.filter((Host.org_id != current_org_id) & Host.id.in_(host_id_list)).all()
+    if host_query:
+        nonexistent_hosts = [host.id for host in host_query]
+        log_host_group_add_failed(logger, nonexistent_hosts, group_id)
+        raise InventoryException(
+            title="Invalid request", detail=f"Could not find an existing host(s) with ID {host_query}."
+        )
```

Comment:
```suggestion
    # Check if the hosts exist in Inventory and correct account
    host_query = Host.query.filter((Host.org_id == current_org_id) & Host.id.in_(host_id_list)).all()
    found_ids_set = {str(host.id) for host in host_query}
    if found_ids_set != set(host_id_list):
        nonexistent_hosts = set(host_id_list) - found_ids_set
        log_host_group_add_failed(logger, nonexistent_hosts, group_id)
        raise InventoryException(
            title="Invalid request", detail=f"Could not find an existing host(s) with ID {nonexistent_hosts}."
        )
```
What do you say about something like this? The original check query would catch hosts in different account, but it wouldn't catch non-existing hosts, which then return error "A group with host list ['290fec21-f2b1-49df-92af-3194c667cac2'] already exists.", because it will pass this and fail on DB addition.

Reply:
Also, can we move this check above the check for already assigned hosts? That way we can return this error message even if we try to add a host from different account which is already assigned to a group.

Reply:
hahaha, that looks a lot like what I had written before changing to this version I committed. I'll change back to what you have suggested